tensorflow_lite_support/python/task/vision/image_classifier_test.py

```diff
@@ -0,0 +1,535 @@
+            r"'file_content', file_name' or 'file_descriptor_meta'\. "
+            r"\[tflite::support::TfLiteSupportStatus='2'\]"):
+      base_options = task_options.BaseOptions(model_file="")
+      options = image_classifier.ImageClassifierOptions(base_options=base_options)
+      image_classifier.ImageClassifier.create_from_options(options)
+
+    # Invalid max results.
+    with self.assertRaisesRegex(
+            Exception,
+            r"INVALID_ARGUMENT: Invalid `max_results` option: value must be != 0 "
+            r"\[tflite::support::TfLiteSupportStatus='2'\]"):
+      base_options = task_options.BaseOptions(model_file=model_file)
+      classifier_options = processor_options.ClassificationOptions(
+        max_results=0)
+      options = image_classifier.ImageClassifierOptions(
+        base_options=base_options,
+        classifier_options=classifier_options)
+      image_classifier.ImageClassifier.create_from_options(options)
+
+    # Invalid number of threads.
+    with self.assertRaisesRegex(
+            Exception,
+            r"INVALID_ARGUMENT: `num_threads` must be greater than 0 or equal to -1. "
+            r"\[tflite::support::TfLiteSupportStatus='2']"):
+      base_options = task_options.BaseOptions(model_file=model_file, num_threads=-2)
+      options = image_classifier.ImageClassifierOptions(base_options=base_options)
+      image_classifier.ImageClassifier.create_from_options(options)
+
+  @parameterized.parameters(
+    (_MODEL_FLOAT, ['foo'], ['bar']),
+    (_MODEL_QUANTIZED, ['foo'], ['bar']),
+    (_MODEL_AUTOML, ['foo'], ['bar']),
+  )
+  def test_combined_whitelist_and_blacklist(self, model_name,
+                                            class_name_whitelist,
+                                            class_name_blacklist):
+    # Get the model path from the test data directory
+    model_file = test_util.get_test_data_path(model_name)
+
+    # Fails with combined whitelist and blacklist
+    with self.assertRaisesRegex(
+            Exception,
+            r"INVALID_ARGUMENT: `class_name_whitelist` and `class_name_blacklist` "
+            r"are mutually exclusive options. "
+            r"\[tflite::support::TfLiteSupportStatus='2'\]"):
+      base_options = task_options.BaseOptions(model_file=model_file)
+      classifier_options = processor_options.ClassificationOptions(
+        class_name_whitelist=class_name_whitelist,
+        class_name_blacklist=class_name_blacklist)
+      options = image_classifier.ImageClassifierOptions(
+        base_options=base_options,
+        classifier_options=classifier_options)
+      image_classifier.ImageClassifier.create_from_options(options)
+
+  @parameterized.parameters(
+    (_MODEL_FLOAT, 3, False),
+    (_MODEL_QUANTIZED, 3, False),
+    (_MODEL_AUTOML, 3, False),
+    (_MODEL_FLOAT, 3, True),
+    (_MODEL_QUANTIZED, 3, True),
+    (_MODEL_AUTOML, 3, True),
+  )
+  def test_classify_model(self, model_name, max_results, with_bounding_box):
```

Comment:
@kinaryml 
Please split to `test_classify_model` (without bounding box) and `test_classify_model_with_bounding_box` for better readability.

@lu-wang-g @mbrenon @ziyeqinghan 
I think it's fine to test with just one TFLite model file (e.g. only with the float model) as it's the responsibility of the underlying C++ layer to ensure that Task Library works correctly with both float and quantized model. WDYT?

Reply:
Updated the file to use two separate test cases (classification w/ and w/o the bounding box).